lib/core/engine_clang.cpp

```diff
@@ -339,45 +332,58 @@ namespace
   {
     std::vector<std::string>
       args{
-        "-Wfatal-errors",
         "-iquote", ".",
         "-x", "c++-header",
         "-I", internal_dir_
       };
 
-    if (env_detector_.on_windows())
+    if (UseInternalTemplight)
     {
-      args.push_back("-fno-ms-compatibility");
-      args.push_back("-U_MSC_VER");
-    }
+      args.push_back("-Wfatal-errors");
 
-    if (!cpp_standard_set(config_.extra_clang_args))
-    {
-      args.push_back("-std=c++0x");
-    }
+      if (env_detector_.on_windows())
+      {
+        args.push_back("-fno-ms-compatibility");
+        args.push_back("-U_MSC_VER");
+      }
+
+      if (!cpp_standard_set(config_.extra_clang_args))
+      {
+        args.push_back("-std=c++0x");
+      }
+
+      if (!max_template_depth_set(config_.extra_clang_args))
+      {
+        args.push_back(set_max_template_depth(256));
+      }
   
-    if (!max_template_depth_set(config_.extra_clang_args))
-    {
-      args.push_back(set_max_template_depth(256));
-    }
+      const std::vector<std::string> include_path =
+        determine_include_path(clang_path_, env_detector_, logger_);
+      args.reserve(args.size() + include_path.size());
+      for (const std::string& p : include_path)
+      {
+        args.push_back("-I" + p);
+      }
 
-    const std::vector<std::string> include_path =
-      determine_include_path(clang_path_, env_detector_, logger_);
-    args.reserve(args.size() + include_path.size());
-    for (const std::string& p : include_path)
+      args.insert(
+        args.end(),
+        config_.extra_clang_args.begin(),
+        config_.extra_clang_args.end()
+      );
+    }
+    else if (config_.extra_clang_args.size() > 1)
     {
-      args.push_back("-I" + p);
+      args.insert(
+        args.end(),
+        config_.extra_clang_args.begin() + 1,
+        config_.extra_clang_args.end()
+      );
     }
 
-    args.insert(
-      args.end(),
-      config_.extra_clang_args.begin(),
-      config_.extra_clang_args.end()
-    );
-
     return args;
   }
 
+  template <bool UseInternalTemplight>
```

Comment:
I wouldn't make this a template parameter. It's only used in run time contexts and as far as I can see the only benefit is that it makes the usage of factory_for easy.

How about making UseInternalTemplate a runtime member of engine_clang and solving the factory_for issue by adding a new `Args...` parameter to factory_for and passing that to the engine's constructor?

Something like:
```cpp
  template <class Engine, class... Args>
  engine_factory factory_for(const Args&... args)
  {
    return
      [&](
        const data::config& config_,
        const std::string& internal_dir_,
        const std::string& env_filename_,
        iface::environment_detector& env_detector_,
        iface::displayer& displayer_,
        logger* logger_
      )
      {
        return
          std::unique_ptr<iface::engine>(
            new Engine(
              config_,
              internal_dir_,
              env_filename_,
              env_detector_,
              displayer_,
              logger_,
              args...
            )
          );
      };
  }
```

What do you think?